File: backend/Domain/Entities/WorkRequest.cs
using WorkRequestTracker.Domain.Enums;

namespace WorkRequestTracker.Domain.Entities;

public class WorkRequest
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string ClientName { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Priority Priority { get; set; }
    public WorkRequestStatus Status { get; set; }
    public DateTimeOffset DueDate { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
    public List<Note> Notes { get; set; } = new();
}

File: frontend/types/workRequest.ts
export type Priority = 'Low' | 'Medium' | 'High';
export type WorkRequestStatus = 'New' | 'InProgress' | 'Blocked' | 'Completed';

export interface Note {
  id: string;
  content: string;
  createdAt: string;
}

export interface WorkRequestSummary {
  id: string;
  title: string;
  clientName: string;
  priority: Priority;
  status: WorkRequestStatus;
  dueDate: string;
  createdAt: string;
}

export interface WorkRequest extends WorkRequestSummary {
  description: string;
  updatedAt: string;
  notes: Note[];
}

export interface CreateWorkRequestPayload {
  title: string;
  clientName: string;
  description: string;
  priority: Priority;
  status: WorkRequestStatus;
  dueDate: string;
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
  errors?: string[];
  pagination?: {
    page: number;
    pageSize: number;
    totalCount: number;
    totalPages: number;
  };
}
